src/main/java/com/iridium/iridiumteams/gui/BlockValueGUI.java

```diff
@@ -0,0 +1,128 @@
+                .replace("%blockType%", guiTitle)
+                .replace("%page%", String.valueOf(getPage()))
+                .replace("%max_pages%", String.valueOf(maxPages))
+        ));
+        addContent(inventory);
+        return inventory;
+    }
+
+    @Override
+    public void addContent(Inventory inventory) {
+        super.addContent(inventory);
+
+        if(blockValueType == BlockValueType.BLOCK) {
+            for (Map.Entry<XMaterial, BlockValues.ValuableBlock> entry : iridiumTeams.getBlockValues().blockValues.entrySet().stream().filter(entry -> entry.getValue().page == getPage()).collect(Collectors.toList())) {
+
+                List<String> lore = new ArrayList<>();
+                lore.add(iridiumTeams.getBlockValues().valueLore.replace("%block_value%", String.valueOf(entry.getValue().value)));
+                lore.add(iridiumTeams.getBlockValues().teamValueLore
+                        .replace("%total_blocks%", String.valueOf(
+                                iridiumTeams.getTeamManager().getTeamBlock(team, entry.getKey()).getAmount()))
+                        .replace("%total_block_value%", String.valueOf(
+                                iridiumTeams.getTeamManager().getTeamBlock(team, entry.getKey()).getAmount() * entry.getValue().value
+                        )));
+
+                inventory.setItem(entry.getValue().slot, ItemStackUtils.makeItem(entry.getKey(), 1, entry.getValue().name, lore));
+            }
+        }
+
+        if(blockValueType == BlockValueType.SPAWNER) {
+            for (Map.Entry<EntityType, BlockValues.ValuableBlock> entry : iridiumTeams.getBlockValues().spawnerValues.entrySet().stream().filter(entry -> entry.getValue().page == getPage()).collect(Collectors.toList())) {
+
+                List<String> lore = new ArrayList<>();
+                lore.add(iridiumTeams.getBlockValues().valueLore.replace("%block_value%", String.valueOf(entry.getValue().value)));
+                lore.add(iridiumTeams.getBlockValues().teamValueLore
+                        .replace("%total_blocks%", String.valueOf(
+                                iridiumTeams.getTeamManager().getTeamSpawners(team, entry.getKey()).getAmount()))
+                        .replace("%total_block_value%", String.valueOf(
+                                iridiumTeams.getTeamManager().getTeamSpawners(team, entry.getKey()).getAmount() * entry.getValue().value
+                        )));
+
+                String itemName = entry.getKey().name().toUpperCase() + "_SPAWN_EGG";
+                Optional<XMaterial> material = XMaterial.matchXMaterial(itemName);
+                if(!material.isPresent()) material = XMaterial.matchXMaterial("SPAWNER");
+                XMaterial item = material.get();
+
+                inventory.setItem(entry.getValue().slot, ItemStackUtils.makeItem(item, 1, entry.getValue().name, lore));
+            }
+        }
+    }
```

Comment:
Instead of putting this stuff here, we should consider implementing this in the getItemStack method, then the super.addContent will implement this all for us

Reply:
If you look at the Members GUI: https://github.com/Iridium-Development/IridiumTeams/blob/master/src/main/java/com/iridium/iridiumteams/gui/MembersGUI.java

You can see we dont override the addContent method, we implement the getItemStack and  getPageObjects methods.

So I think here, all we need to do is implement the getItemStack method which will get us an itemstack from a ValuableBlock object